backend/main.py

```
@@ -0,0 +1,550 @@
+            llm_used=llm_used
+        )
+
+    except HTTPException:
+        raise
+    except Exception as e:
+        processing_time = time.time() - start_time
+        logger.error(f"Error processing query after {processing_time:.3f}s: {e}")
+        raise HTTPException(
+            status_code=500,
+            detail="Internal server error - please try again later"
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)